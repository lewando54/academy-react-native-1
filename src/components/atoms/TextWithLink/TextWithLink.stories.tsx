/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react'

import { MemoryRouter } from 'react-router-native'
import TextWithLink from './TextWithLink'
import { create } from 'react-test-renderer'

interface ITextWithLinkMockupProps {
    paragraphText: string
    href: string
    anchorText: string
}

function RouterWrapTextWithLink({paragraphText, href, anchorText}: ITextWithLinkMockupProps){

    return (
        <MemoryRouter initialEntries={['/', '/test']}>
            <TextWithLink paragraphText={paragraphText} href={href} anchorText={anchorText} testId='test'/>
        </MemoryRouter>
    )
}

const meta = {
    component: RouterWrapTextWithLink,
    argTypes: {
        paragraphText: { control: 'string' },
        href: { control: 'string' },
        anchorText: { control: 'string' }
    }
} satisfies Meta<typeof RouterWrapTextWithLink>

export default meta
type Story = StoryObj<typeof meta>

export const Default: Story = {
    args: {
        paragraphText: 'If you want cookies',
        href: '/test',
        anchorText: 'click me!'
    }
}