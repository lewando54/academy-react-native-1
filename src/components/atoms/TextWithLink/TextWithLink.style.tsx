import { StyleSheet } from 'react-native'

export default StyleSheet.create({
    div:{
        display: 'flex',
        gap: 4,
        lineHeight: 20,
        alignItems: 'center',
        justifyContent: 'center',
        flexDirection: 'row'
    },
    text:{
        color: 'rgb(107 114 128)',
        fontSize: 14
    },
    link:{
        color: 'rgb(107 114 128)',
        textDecorationLine: 'underline',
        fontSize: 14
    }
})