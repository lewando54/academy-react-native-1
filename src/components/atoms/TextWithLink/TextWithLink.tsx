import React from 'react'
import { useNavigate } from 'react-router-dom'
import textWithLinkStyle from './TextWithLink.style'
import { View, Pressable, Text } from 'react-native'

interface TextWithLinkProps {
    paragraphText: string
    href: string
    anchorText: string
    testId: string
    anchorTagTestId?: string
}

function TextWithLink({paragraphText, href, anchorText, testId}: TextWithLinkProps): React.ReactElement {
    const navigate = useNavigate()

    const onClick = () => {
        navigate(href)
    }
    
    return (
        <View style={textWithLinkStyle.div} testID={testId}>
            <Text style={textWithLinkStyle.text}>
                {paragraphText}
            </Text>
            <Pressable onPress={onClick}>
                <Text style={[textWithLinkStyle.text, textWithLinkStyle.link]}>
                    {anchorText}
                </Text>
            </Pressable>
        </View>
    )
}

export default TextWithLink