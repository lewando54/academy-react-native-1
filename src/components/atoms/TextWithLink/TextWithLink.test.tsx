import * as renderer from 'react-test-renderer'
import TextWithLink from './TextWithLink'
import { fireEvent, render } from '@testing-library/react-native'
import { MemoryRouter, Route, Routes } from 'react-router-native'
import { Text } from 'react-native'

describe('Text With Link', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<MemoryRouter><TextWithLink paragraphText='test' anchorText='click me' href='/' testId='test' anchorTagTestId='atest' /></MemoryRouter>)
    expect(tree).toMatchSnapshot()
  })

  it('should render paragraphText correctly', () => {
    const { getByText } = render(<MemoryRouter><TextWithLink paragraphText='testingP' anchorText='click me' href='/' testId='test' anchorTagTestId='atest' /></MemoryRouter>)
    expect(getByText('testingP')).toBeTruthy()
  })

  it('should render anchor tag text correctly', () => {
    const { getByText } = render(<MemoryRouter><TextWithLink paragraphText='testingP' anchorText='click me' href='/' testId='test' anchorTagTestId='atest' /></MemoryRouter>)
    expect(getByText('click me')).toBeTruthy()
  })

  it('should navigate in app properly after click', () => {
    const href = '/test'
    const { getByText } = render(
      <MemoryRouter initialEntries={['/']}>
        <Routes>
          <Route path='/' element={<TextWithLink paragraphText='testingP' anchorText='click me' href='/test' testId='test' />} />
          <Route path='/test' element={<Text>Correct!</Text>} />
        </Routes>
    </MemoryRouter>
    )

    fireEvent.press(getByText('click me'))

    expect(getByText('Correct!')).toBeTruthy()
  })
})
