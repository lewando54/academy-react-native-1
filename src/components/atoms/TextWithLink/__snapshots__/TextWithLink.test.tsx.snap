// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text With Link should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "row",
      "gap": 4,
      "justifyContent": "center",
      "lineHeight": 20,
    }
  }
  testID="test"
>
  <Text
    style={
      {
        "color": "rgb(107 114 128)",
        "fontSize": 14,
      }
    }
  >
    test
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <Text
      style={
        [
          {
            "color": "rgb(107 114 128)",
            "fontSize": 14,
          },
          {
            "color": "rgb(107 114 128)",
            "fontSize": 14,
            "textDecorationLine": "underline",
          },
        ]
      }
    >
      click me
    </Text>
  </View>
</View>
`;
